modules/renter/contractor/journal.go

```diff
@@ -0,0 +1,332 @@
+package contractor
+
+// The contractor achieves efficient persistence using a JSON transaction
+// journal. It enables efficient ACID transactions on JSON objects.
+//
+// The journal represents a single JSON object, containing all of the
+// contractor's persisted data. The object is serialized as an "initial
+// object" followed by a series of update sets, one per line. Each update
```

Comment:
If you run into a corrupted update, everything after it is also corrupt because the process is linear. Imagine that you've got two appends and one is corrupt. Then your second append is going to give you an improper state.

If you hit a corrupt entry, nothing after it should be assumed to have sync'd. You should stop reading, even if the later checksums match.